Fix the build on non amd64/i386 platforms by patching the I/O code so we
only call outb() if HAVE_I386_SET_IOPERM or HAVE_LIBAMD64 are defined.

Index: src/iodrv/io-unix-access.c
--- src/iodrv/io-unix-access.c.orig
+++ src/iodrv/io-unix-access.c
@@ -349,7 +349,7 @@ static inline uint8_t vice_inb(uint16_t port)
     return inb_p(port);
 }
 #  else
-#    ifdef __OpenBSD__
+#    if defined(__OpenBSD__) && (defined(HAVE_LIBAMD64) || defined(HAVE_I386_SET_IOPERM))
 static inline void vice_outb(uint16_t port, uint8_t val)
 {
     outb(port, val);
